### PR TITLE
Add a simple initial functional test for carousel-v2.

### PR DESCRIPTION
**Pull request by sparhami**
- Add eslint carve out for fn_test directories in extensions to allow `beforeEach`, etc.
- Fix findElement(s)Xpath
- Add an simple test for carousel based on a manual test file

### CODE REVIEW

File: extensions/amp-carousel/0.2/fn_test/test-carousel.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import 'babel-regenerator-runtime';
```
Comment (cvializ) [inline extensions/amp-carousel/0.2/fn_test/test-carousel.js]: @estherkim and I discussed folder layout and we settled on putting the end-to-end tests' JS files in the `test/e2e/` folder, and the HTML files in `amphtml/fixtures/` (maybe `amphtml/fixtures/e2e`?). This has the benefits of decoupling the test JS from the extension JS and deemphasizing e2e tests from being only "tests for this AMP extension" and emphasizing that they should be "tests for this AMP user flow". What do you think? Glad to have a quick VC sync to discuss ideas.

File: testing/e2e/selenium-webdriver-controller.js [hunk 2]
```diff
@@ -109,11 +109,10 @@ export class SeleniumWebDriverController {
    * @override
    */
   async findElementXPath(xpath) {
-    const {browser} = this;
     const byXpath = By.xpath(xpath);
 
-    await browser.wait(until.elementLocated(byXpath));
-    const webElement = await browser.findElement(byXpath);
+    await this.driver.wait(until.elementLocated(byXpath));
```
Comment (cvializ) [inline testing/e2e/selenium-webdriver-controller.js]: Thanks, forgot to port this change.

File: extensions/amp-carousel/0.2/fn_test/test-carousel.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    return await controller.findElementXPath(
+        `//amp-carousel//div[contains(@class, "${slottedClass}")][${n + 1}]`);
+  }
+
+  beforeEach(async() => {
+    controller = env.controller;
+
+    // Enable the amp-carousel-v2 and layers experiments.
+    await controller.navigateTo(
+        'http://localhost:8000/test/manual/amp-carousel-0-2/enable-experiment.html');
```
[the remark this answers is not in the document]
Comment (sparhami) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: Yeah, that would be great!

File: extensions/amp-carousel/0.2/fn_test/test-carousel.js (hunk 1 above)
Comment (sparhami) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: I would strongly prefer that tests for a single component are beside the code they are testing. If we want to have functional tests that test multiple components together, I agree a centralized location would be the best place. Having the tests in another folder just puts them out of sight, making a developer less likely to remember to add tests as they make changes.

Comment (estherkim) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: Makes sense to me re: testing a single component. Can we change fn_test to e2e-test or e2e?

File: extensions/amp-carousel/0.2/fn_test/test-carousel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (cvializ) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: kk keep this for now and I'll cc you when I add that

File: extensions/amp-carousel/0.2/fn_test/test-carousel.js (hunk 1 above)
Comment (sparhami) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: Changed the folder name to `e2e-test`.

Comment (estherkim) [reply extensions/amp-carousel/0.2/fn_test/test-carousel.js]: Ah wait! Can we change it to `test-e2e` so it stays close to the other `test` folder? Sorry my bad!

**Pull request merged**